From 265d4bcbae9ae0f9594d5fcc661842dfe434ceb7 Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Wed, 17 Jul 2019 19:56:02 +0800
Subject: [PATCH] drm/rockchip: driver: set loader_protect when in kernel logo
 on

protect restore_fbdev_mode in kernel logo on until userspace
power on.

Change-Id: I561d9eaa3a931471cdc81b41f6c0a7db28588ba0
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_drv.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
index 6fe9882ac17e..dae7ab74532e 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
@@ -223,6 +223,7 @@ void rockchip_free_loader_memory(struct drm_device *drm)
 	free_reserved_area(start, end, -1, "drm_logo");
 	kfree(logo);
 	private->logo = NULL;
+	drm->mode_config.loader_protect = false;
 }
 
 static int init_loader_memory(struct drm_device *drm_dev)
@@ -963,6 +964,7 @@ static void show_loader_logo(struct drm_device *drm_dev)
 	rockchip_free_loader_memory(drm_dev);
 	drm_atomic_state_put(old_state);
 
+	mode_config->loader_protect = true;
 	drm_modeset_unlock_all(drm_dev);
 	return;
 err_free_old_state:
-- 
2.35.3

